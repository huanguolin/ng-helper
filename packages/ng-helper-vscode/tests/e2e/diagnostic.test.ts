import { expect } from 'chai';
import { describe, it, before } from 'mocha';
import * as vscode from 'vscode';

import { APP_PAGES_P5_HTML_PATH, CHECK_TIME_DIRECTIVE_JS_PATH } from '../testConstants';
import { activate, setupChaiSnapshotPlugin, sleep } from '../testUtils';

describe('Diagnostic', () => {
    setupChaiSnapshotPlugin();

    before(async () => {
        await activate(1);
    });

    it('html diagnostic', async () => {
        await testSemantic(APP_PAGES_P5_HTML_PATH);
    });

    it('inline html diagnostic', async () => {
        await testSemantic(CHECK_TIME_DIRECTIVE_JS_PATH);
    });
});

async function testSemantic(filePath: string, waitSeconds?: number) {
    const uri = vscode.Uri.file(filePath);

    // show the document
    await vscode.window.showTextDocument(uri);
    if (waitSeconds) {
        await sleep(waitSeconds * 1000);
    }

    const diagnostics = vscode.languages.getDiagnostics(uri);

    // assert
    expect(diagnostics.length).to.be.greaterThan(0);
    expect(diagnostics).toMatchSnapshot();
}
