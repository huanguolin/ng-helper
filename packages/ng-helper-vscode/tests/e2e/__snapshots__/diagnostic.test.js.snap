// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Diagnostic html diagnostic 1`] = `
Array [
  Object {
    "code": undefined,
    "message": "Expression expected",
    "range": Array [
      Object {
        "character": 2,
        "line": 2,
      },
      Object {
        "character": 3,
        "line": 2,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "Expression expected",
    "range": Array [
      Object {
        "character": 70,
        "line": 4,
      },
      Object {
        "character": 70,
        "line": 4,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "Identifier expected",
    "range": Array [
      Object {
        "character": 36,
        "line": 6,
      },
      Object {
        "character": 37,
        "line": 6,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "Expression expected",
    "range": Array [
      Object {
        "character": 23,
        "line": 8,
      },
      Object {
        "character": 24,
        "line": 8,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "Literal is not callable",
    "range": Array [
      Object {
        "character": 18,
        "line": 8,
      },
      Object {
        "character": 19,
        "line": 8,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "Unexpected character: &",
    "range": Array [
      Object {
        "character": 10,
        "line": 10,
      },
      Object {
        "character": 11,
        "line": 10,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "Expression expected",
    "range": Array [
      Object {
        "character": 27,
        "line": 13,
      },
      Object {
        "character": 27,
        "line": 13,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "')' expected",
    "range": Array [
      Object {
        "character": 31,
        "line": 15,
      },
      Object {
        "character": 31,
        "line": 15,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "Expression expected",
    "range": Array [
      Object {
        "character": 54,
        "line": 15,
      },
      Object {
        "character": 54,
        "line": 15,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
]
`;

exports[`Diagnostic inline html diagnostic 1`] = `
Array [
  Object {
    "code": undefined,
    "message": "Expression expected",
    "range": Array [
      Object {
        "character": 2,
        "line": 14,
      },
      Object {
        "character": 3,
        "line": 14,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "Expression expected",
    "range": Array [
      Object {
        "character": 70,
        "line": 16,
      },
      Object {
        "character": 70,
        "line": 16,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "Identifier expected",
    "range": Array [
      Object {
        "character": 36,
        "line": 18,
      },
      Object {
        "character": 37,
        "line": 18,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "Expression expected",
    "range": Array [
      Object {
        "character": 23,
        "line": 20,
      },
      Object {
        "character": 24,
        "line": 20,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "Literal is not callable",
    "range": Array [
      Object {
        "character": 18,
        "line": 20,
      },
      Object {
        "character": 19,
        "line": 20,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "Unexpected character: &",
    "range": Array [
      Object {
        "character": 10,
        "line": 22,
      },
      Object {
        "character": 11,
        "line": 22,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "Expression expected",
    "range": Array [
      Object {
        "character": 27,
        "line": 25,
      },
      Object {
        "character": 27,
        "line": 25,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "')' expected",
    "range": Array [
      Object {
        "character": 31,
        "line": 27,
      },
      Object {
        "character": 31,
        "line": 27,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
  Object {
    "code": undefined,
    "message": "Expression expected",
    "range": Array [
      Object {
        "character": 54,
        "line": 27,
      },
      Object {
        "character": 54,
        "line": 27,
      },
    ],
    "severity": "Error",
    "source": "[ng-helper]",
  },
]
`;
